src/Cluster.php

```diff
@@ -33,6 +38,7 @@ class Cluster {
      * @param Coordinate[optional] $coord
      */
     public function __construct(Coordinate $coord, Coordinate $coord2 = null /*, ... */) {
+        $this->coordinates[] = $coord;
```

Comment:
Can you make it something like:

```php
/**
 * @var bool
 */
protected $saveCoordinates = false;

/**
 * Save all coordinates per cluster.
 *
 * Note that while it allows you to retrieve all Coordinate objects
 * in a cluster, this does not scale. At some point, if you keep
 * adding coordinates into clusters, you'll run out of memory
 * because we're saving all those coordinates.
 * If you don't need the exact information of coordinates in a
 * cluster, leave this disabled.
 *
 * @param bool $save
 */
public function saveCoordinates($save = true) {
    $this->saveCoordinates = $save;
}

    ...
    if ($this->saveCoordinates) {
        $this->coordinates[] = $coord;
    }
    ...
```